### PR TITLE
Add repo and citation stats to README, docs; restructure README

### PR DESCRIPTION
**Pull request by jeffjennings**
### Description

**Motivation:** Increase the transition of astropy users to contributors. Give users a quick, clear view of how heavily the package is used in astronomy research, and how actively the codebase is being developed, in order to give them a sense that their contributions have a big impact. Give their contributions more visibility to increase recognition.

**Changes:** This PR does three closely related things: 
- Adds a 'user stats' image to the README and to the main index in the docs, showing statistics for the astropy/astropy repo and astropy papers
- Adds an 'astropy impact and health' page to the docs (linked from the user stats images) with plots and some narrative on trends in repo and citation stats. 
- Updates the README to be more succinct and more similar in style to the NumPy README (with the goal of being a bit more useful as a quick reference and more engaging to users)
- All changes are viewable in a local build of the docs. Or view the content of the README [here](https://github.com/jeffjennings/astropy/tree/readme_user_dashboard?tab=readme-ov-file), of the 'astropy impact and health' page [here](https://github.com/jeffjennings/astropy/blob/readme_user_dashboard/docs/impact_health.rst), and of the docs main index in the image at the end of this comment.

**Implementation details:** 
- Neither the user stats image nor the figures are generated in astropy. They're all linked to images generated by a repo I made for this, [repo_stats](https://github.com/astropy/repo_stats). That code queries GitHub's GraphQL API and the NASA ADS API daily to update and cache the necessary stats, then generates timestamped images. 
- ~I'm ok to leave that repo where it is, but if necessary I could instead transfer ownership of it to the astropy organization. I don't have any organization permissions so would need to coordinate with someone to set up the github actions workflow secrets and update the image links.~ [Update: the repo is now astropy/repo_stats]
- The images generated by `repo_stats` could be generated for any repo and set of papers (e.g. for coordinated packages). A .json parameter file just needs to be updated with the repo owner/name and paper bibcodes. If people are interested in this, please let me know for which packages/papers and I could open a PR in the relevant repo.
- If there are other plots you'd like to have generated and/or added to the 'astropy impact and health' docs page, I'd suggest that go in a separate issue. If you'd like the underlying data or image files, the most recent cache of stats and images can always be downloaded from the `cache` orphan branch in `repo_stats`.
- If you have any other questions about or issues with the implementation in `repo_stats`, let me know!

**Related issues:** #16560 (addresses but may not close), [Astropy Project issue 188](https://github.com/astropy/astropy-project/issues/188) (closes, I'd say - I'm not able to link that issue to this PR directly without permissions)

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

Screenshot 2024-08-13 at 1 13 23 PM

### CODE REVIEW

File: README.rst
Comment (pllim) [inline README.rst:10]: Thanks.

I am not sure about having this right at the top. If I were someone new wondering what Astropy is, these stats don't mean anything to me, I would want to read "about" first.

Isn't this more appropriate under "contributing"?

Comment (jeffjennings) [reply README.rst:10]: My thought was that the 'user stats image' is best near the top, as it shows aspects of the status of the repo, similar to the badges. The metrics it lists are general I'd say - basic git things (contributor, issue, PR counts) and citations (which probably most people who use astropy are familiar with). And since it's kind of a big image, I think it would break up the readme and make it less readable if moved to a lower section.

Comment (mhvk) [reply README.rst:10]: For `README.rst`, I agree with you that in a way it belongs in the line of badges. Indeed, one might want to put it *above* the badges... (but for the main docs page, I feel differently)

File: docs/impact_health.rst
Comment (mhvk) [inline docs/impact_health.rst:6]: Since this is effectively a kind of abstract, maybe add a sentence that we'd like help, e.g., "Hence, we welcome any help that you can give!"

File: docs/impact_health.rst
Comment (mhvk) [inline docs/impact_health.rst:5]: The rest of our docs stick with a line length limit. I think that makes things easier to review.

**Review by mhvk — COMMENTED** [review]
I very much like your addition with the quick contributors/citation statistics! (And love how you managed to get the logo to fit in there!) While I think having it near the top for the `README` is fine, for the main astropy page I think it should be at the bottom - that page is not so big, and it is most important to be able to just click on "user guide", etc.

File: README.rst
Comment (bsipocz) [inline README.rst:79]: I think this PR should be held back from merging until all these linked images here and below can be sourced from an astopy-org-hosted repo that has a bus factor/access >>1 person.

File: README.rst
Comment (jeffjennings) [reply README.rst]: Great, I moved the image above the badges in `README.rst` and to the bottom of the main docs page in [6a6dadd](https://github.com/astropy/astropy/pull/16822/commits/6a6dadd3c4b616ec01b62b443eb3ce7f9943c9a8).

File: docs/impact_health.rst
Comment (jeffjennings) [reply docs/impact_health.rst]: Good idea! I added the line in [0e164df](https://github.com/astropy/astropy/pull/16822/commits/0e164df73e95118d352f286dd92bb1603a2329d2)

File: docs/impact_health.rst
Comment (jeffjennings) [reply docs/impact_health.rst]: Line limit applied in [0e164df](https://github.com/astropy/astropy/pull/16822/commits/0e164df73e95118d352f286dd92bb1603a2329d2)

File: README.rst
Comment (jeffjennings) [reply README.rst:80]: Great, that makes sense. It seems the easiest way would be to duplicate the repository to within the astropy organization; following the GitHub [guide](https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository), can someone with the permission to create a new (empty) repo in the organization with the name `repo_stats` do that and let me know? I can then submit a PR there with a mirror of the current repo with updated image hyperlinks. I can also coordinate with that person to add the needed GitHub Actions secrets.

Comment (pllim) [reply README.rst:80]: If you give me admin access to your repo, I can probably do a straight transfer. We can coordinate privately if you want.

Comment (jeffjennings) [reply README.rst:80]: Ok yes I've sent you an admin invite, and I'll open a PR in the new repo when it's transferred to update the links.

Comment (pllim) [reply README.rst]: Transfer is complete. Thanks, Jeff! Please refer to https://github.com/astropy/repo_stats now. 🙇‍♀️

Comment (jeffjennings) [reply README.rst]: Great, thank you!

**Comment by jeffjennings** [comment]
> I very much like your addition with the quick contributors/citation statistics! (And love how you managed to get the logo to fit in there!) While I think having it near the top for the `README` is fine, for the main astropy page I think it should be at the bottom - that page is not so big, and it is most important to be able to just click on "user guide", etc.

Thanks Marten!

File: README.rst
Comment (pllim) [inline README.rst:1]: Is it just me or GitHub is not rendering this file? It used to render in an earlier stage of the PR. Now I just see the raw RST stuff. "View file" is supposed to render, right?

File: docs/index.rst
Comment (pllim) [inline docs/index.rst:126]: Our doc supports both light/dark mode. The light version looks out of place in dark mode. Is there a way to make this image auto-adjust to light/dark or is this the same problem as https://github.com/matplotlib/matplotlib/issues/25983 ?

File: README.rst
Comment (pllim) [inline README.rst:84]: I don't understand why logo is light but the user stats is dark here.

File: docs/impact_health.rst
Comment (pllim) [inline docs/impact_health.rst:1]: Since the plots are dynamic but not the text, how do we ensure the text actually is in sync with plots going forward? Do we need a perpetually open issue to revisit this once in a while?

Also I think "astropy" should be formatted `astropy` (with double backticks for RST), please see https://docs.astropy.org/en/latest/development/style-guide.html#astropy-style-guide .

File: README.rst
Comment (pllim) [reply README.rst:83]: Doesn't look too good in light mode. I think we need versions of both images that would look okay in either. Also the logo can be smaller. Thanks.

Screenshot 2024-08-16 095814

File: docs/index.rst
Comment (jeffjennings) [reply docs/index.rst]: I had to make the image background of the transparent; updated version being used as of 
b51fb17
Screenshot 2024-08-16 at 11 52 15 AM

File: README.rst
Comment (jeffjennings) [reply README.rst]: the logo file has a transparent background; maybe it was named 'light' because the text is white. 

The logo image is now the gray text version from the docs files - https://github.com/astropy/astropy/blob/main/docs/_static/astropy_banner.svg 

I'd prefer not to resize that file and have a second version of it floating around, so I think the size is ok.

Example:
https://github.com/jeffjennings/astropy/tree/readme_user_dashboard?tab=readme-ov-file

File: docs/impact_health.rst
Comment (jeffjennings) [reply docs/impact_health.rst:1]: The text is fairly general ('the package could use more maintainers'), but the somewhat specific parts commenting on trends (no mention of numbers) will probably be true for a long time, as they've been true over the lifetime of the codebase according to the plots. There could be an issue to periodically check the generated plots on the page, but I don't think it's needed.

File: docs/index.rst
Comment (jeffjennings) [reply docs/index.rst]: The README looks fine in both light and dark, but my local build of the docs shows a white background around the image in dark mode; I don't know why, I'm guessing sphinx doesn't render transparency and instead sets it to white (as the RGB values of transparency are 255, 255, 255, the same as white; the A in RGBA provides transparency information). I don't have a good workaround for that.

File: README.rst
Comment (pllim) [inline README.rst:18]:
```suggestion
* `Astropy developers mailing list <https://groups.google.com/g/astropy-dev>`_
```

File: docs/impact_health.rst
Comment (pllim) [inline docs/impact_health.rst:32]: Hmm interesting that issues and PRs open/close kinda track each other, respectively. What does this mean?

Also, will your conclusion hold if I tell you we have a stale bot to auto close stale PRs? See https://docs.astropy.org/en/latest/development/maintainers/maintainer_workflow.html#stale-policies

**Review by pllim — COMMENTED** [review]
I don't know how I feel about removal of TOC and Codespaces "note" but I also don't have strong objection.

How relevant are these documents to your content here? Should we also link to them?

* https://astropy-report.orgmycology.com/
* https://astropy-dei.orgmycology.com/

**Comment by pllim** [comment]
Just a few more comments, otherwise LGTM. Thanks!

File: docs/impact_health.rst
Comment (jeffjennings) [reply docs/impact_health.rst:32]: I assume opens and closes track each other for both issues and PRs because maybe a majority of issues can be closed fairly quickly by new PRs, which themselves don't need to be open long. 

https://github.com/jeffjennings/repo_stats/blob/89408b954d846d5a8e0bee169d08be7915f5b95c/repo_stats/git_metrics.py#L287 has a list of bots whose commits I ignore when computing commit statistics (if you know of others, please add them). Since that list already has `github-actions` and it looks like the stale bot uses this ID (e.g. https://github.com/astropy/astropy/issues/16435), the stale bot commits are excluded from these plots.

**Comment by jeffjennings** [comment]
> I don't know how I feel about removal of TOC and Codespaces "note" but I also don't have strong objection.
> 
> How relevant are these documents to your content here? Should we also link to them?
> 
> * https://astropy-report.orgmycology.com/
> * https://astropy-dei.orgmycology.com/

Yeah I thought the table of contents was excessive for the small readme, and the codespaces note seemed unnecessary; just trying to cut down on the text. [83f3165](https://github.com/astropy/astropy/pull/16822/commits/83f3165fdca8d438e0c3fe5cca9d58b75b7e83e0) adds the two reports to the link list near the top of the readme.

**Comment by jeffjennings** [comment]
> Just a few more comments, otherwise LGTM. Thanks!

I've addressed everything, thank you for such a careful review catching multiple subtle things!

File: README.rst
Comment (pllim) [inline README.rst:1]: @astrofrog , do you still have concern about how this README would render on PyPI?

File: README.rst
Comment (pllim) [inline README.rst:17]:
```suggestion
* `Astropy users mailing list <https://mail.python.org/mailman/listinfo/astropy>`_
```

File: README.rst
Comment (pllim) [inline README.rst:71]: Turns out the original Codespaces blurb, including the note, was originally added by GitHub Codespaces developer in #14481 . Do we really want to reword the section added by Codespaces expert? Is there a strong reason to do so?

File: README.rst
Comment (pllim) [inline README.rst:20]: I was thinking about adding these under `impact_health.rst`, not necessarily in README like this, but I wonder what others think.

File: docs/impact_health.rst
Comment (pllim) [reply docs/impact_health.rst:32]: Might have to rerun after https://github.com/astropy/repo_stats/issues/6 is resolved?

Comment (jeffjennings) [reply docs/impact_health.rst:32]: That's now closed.

File: README.rst
Comment (jeffjennings) [reply README.rst:20]: Ah sorry I thought you had meant the README. Though yeah I think they're nice here actually; the `impact_health` page is really just about codebase development.

File: README.rst
Comment (jeffjennings) [reply README.rst:71]: The previous text was kind of long and had some legal speak. I think it makes sense in a repo about codespaces, but a big disclaimer isn't needed in every repo where codespaces is used. I think the README should be as short as possible to be useful as a quick reference, with longer comments on things like codespaces better left to a page in the docs if it's needed.

File: README.rst
Comment (pllim) [reply README.rst:20]: I am still not convinced but I will defer to other reviewers.

Comment (adrn) [reply README.rst:20]: No strong opinions on this -- I think it's fine to have in the README, but also makes sense to be in `impact_health.rst`.

File: README.rst
Comment (adrn) [inline README.rst:31]: Some inconsistent word wrapping throughout -- maybe wrap it to 80 chars (I think?) like some other lines here?

File: README.rst
Comment (adrn) [reply README.rst:71]: I think the reword is clearer!

File: docs/impact_health.rst
Comment (adrn) [inline docs/impact_health.rst:2]: I know I said this to you already, but this text is great!

**Review by adrn — APPROVED** [review]
@jeffjennings and I worked on this offline before he submitted, so I didn't feel I needed to review the first few iterations. I think the changes are good and this is in great shape!

File: docs/impact_health.rst
Comment (jeffjennings) [reply docs/impact_health.rst:2]: Thanks!

File: README.rst
Comment (jeffjennings) [reply README.rst]: Fixed in 
868fa26

File: README.rst
Comment (jeffjennings) [reply README.rst]: Sounds good, I've moved the links to these out of the README and into `impact_health.rst`, with some text around them, in [bfd95a7](https://github.com/astropy/astropy/pull/16822/commits/bfd95a71b7e6c7926e4e4e8578ae36259ecb3e7d).

**Comment by jeffjennings** [comment]
I've just addressed all of the new review comments and think this is ready to merge!

File: README.rst
Comment (neutrinoceros) [reply README.rst:1]: I just reviewed the PR with only this issue in mind and I believe we're fine on that front, as long as there's an understanding that both url
```
https://github.com/astropy/repo_stats/blob/cache/cache/astropy_user_stats.png
https://docs.astropy.org/en/latest/impact_health.html
```
are to be preserved for ever.
To be clear, PyPI would render this part of the page **dynamically**, so it won't appear to be out dated on there. In case there are concerns about having dynamic parts in the README.md as rendered on PyPI, I'd also like to point out that there's a clear precedent for it: most badges are dynamic ! For instance, if one visits https://pypi.org/project/astropy/6.0.0/, they'll still see the `pypi` badge showing the most recent release, not 6.0.0

**Review by neutrinoceros — APPROVED** [review]
This looks absolutely gorgeous !
My one minor concern is that I don't know how to check whether the target url for the image (i.e., the url you get redirect to when clicking the image, `https://docs.astropy.org/en/latest/impact_health.html`) is stable, as it's currently 404 (which I assume is expected as long as the PR isn't merged).
In case there's any uncertainty about that, I'm sure it's fine if we merge as is, as long as we make sure to check this before the 7.0.0 release.

File: README.rst
Comment (jeffjennings) [reply README.rst:1]: Thanks Clément! It's a good point that the badges are also dynamic.

**Comment by jeffjennings** [comment]
> This looks absolutely gorgeous ! My one minor concern is that I don't know how to check whether the target url for the image (i.e., the url you get redirect to when clicking the image, `https://docs.astropy.org/en/latest/impact_health.html`) is stable, as it's currently 404 (which I assume is expected as long as the PR isn't merged). In case there's any uncertainty about that, I'm sure it's fine if we merge as is, as long as we make sure to check this before the 7.0.0 release.

Thanks a lot! Yes I think the url should be stable once merged.

**Comment by saimn** [comment]
Do we need to put that in the README in addition to the docs ? Seems a bit weird to have some "last 90 days" stats on the PyPI pages, which are supposed to contain a snapshot of the package at a given time.

Also the text in the image is quite hard to read currently : 
image

I see discussion above about dark mode, and the github repo has white and dark images, why not using those with https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/light-dark.html#theme-dependent-images-and-content ?

**Comment by jeffjennings** [comment]
> Do we need to put that in the README in addition to the docs ? Seems a bit weird to have some "last 90 days" stats on the PyPI pages, which are supposed to contain a snapshot of the package at a given time.
> 
Hi Simon, thanks for your comments. Yes one of the main points of the PR is to increase engagement of users with the code and recognition for their contributions, so I think the stats image is useful in both the README and docs, and I don't think the PyPI inconsistency is reason enough to remove it. As Clément noted, the badges also always show the current statuses on PyPI, not the statuses at the time of the snapshot.

> Also the text in the image is quite hard to read currently : image
> 
> I see discussion above about dark mode, and the github repo has white and dark images, why not using those with https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/light-dark.html#theme-dependent-images-and-content ?

Thanks for noting this. I've changed the text color in the image so that it's more readable on both light and dark backgrounds. (The image being displayed has a transparent background, an update from producing `light` and `dark` versions, because the correct version would need to displayed in both the docs and the README, requiring detection of which color mode the page is in for the markdown file.)

**Comment by saimn** [comment]
Hmm badges are a bit different, their color might be time dependent (for some but not all) but they also give an overview of which tools are used, they give the links to the CI providers, RTD etc.  Even if the color status might not be pertinent they give some useful information about the project, which is the kind of information I can be looking for in the README or on PyPI. And I don't think last 90 stats belong to that category, at least not at the top. Why not putting it on the astropy.org website instead, which is a more general, community-oriented web page ?

> Thanks for noting this. I've changed the text color in the image so that it's more readable on both light and dark backgrounds; it will update on the next cron schedule in <1 day. (The image being displayed has a transparent background, an update from producing light and dark versions, because the correct version would need to displayed in both the docs and the README, requiring detection of which color mode the page is in for the markdown file.)

The transparent background version doesn't seem to work (I see that with Firefox and Chromium), and the red text is a bit aggressive:
image
And you produce light and dark versions of the image so why not use that ? It doesn't need to be the same link as the README one.

**Comment by jeffjennings** [comment]
> Hmm badges are a bit different, their color might be time dependent (for some but not all) but they also give an overview of which tools are used, they give the links to the CI providers, RTD etc. Even if the color status might not be pertinent they give some useful information about the project, which is the kind of information I can be looking for in the README or on PyPI. And I don't think last 90 stats belong to that category, at least not at the top. Why not putting it on the astropy.org website instead, which is a more general, community-oriented web page ?
> 
Adrian was referring to the fact that the stats image will appear on https://pypi.org/project/astropy because it is in the README, and since it is generated dynamically (like the badges), the stats it shows will not be frozen at the snapshot of the PyPI release (also like the badges). But that's fine, there isn't much utility to having frozen stats, and the up-to-date stats don't adversely affect anything on the PyPI page.

Regarding the stats image being in the README, I appreciate your preference, though I will say that this was one the main drivers of the PR (motivated a bit in my last comment and in the PR description), following discussion with some of the maintainers. There has been positive feedback to its presence in the README in the reviews and comments above, including multiple approvals of the changes, so I think this is just a case of not being able to perfectly satisfy everyone unfortunately, and the PR should be moved forward. 

> > Thanks for noting this. I've changed the text color in the image so that it's more readable on both light and dark backgrounds; it will update on the next cron schedule in <1 day. (The image being displayed has a transparent background, an update from producing light and dark versions, because the correct version would need to displayed in both the docs and the README, requiring detection of which color mode the page is in for the markdown file.)
> 
> The transparent background version doesn't seem to work (I see that with Firefox and Chromium), and the red text is a bit aggressive: image And you produce light and dark versions of the image so why not use that ? It doesn't need to be the same link as the README one.

Thank you for catching the rendering issue in the docs. I'm now using the light and dark versions of the image in the docs and the transparent version in the README. The image text is black for the light version and white for the dark version, and a darker gray in the transparent version (which is legible on both light and dark backgrounds).

**Comment by adrn** [comment]
I personally think the visibility this would have in the README outweighs the small inconsistency this would represent when appearing on pypi (i.e. that the image would always be current). As we've all mentioned, there is already a precedent for non-tagged information being displayed on pypi README's via the badges: as you said, the colors don't represent the state of the code at a given pypi version/tag, and they also link to CI runs from `main` instead of the relevant version (and the docs link references "stable", not the version tagged on pypi). This is the case for any project I've seen on pypi and users don't seem to mind. In some sense, this stats image is a little less misleading than the badges because it has a date tag in the image itself.

I think we have 3 options: 
1. Leave this as is and accept that it will constantly update on the pypi page, like the CI badges. 🎉
2. Remove the stats image from the README. 🚀
3. Rework the image generation so that image versions can be tagged and referenced directly (this would then need to be added as a thing to do during the release process). 👀

IMO it is beneficial to have this in the README because that is where many people land and get a first look at the package (especially potential contributors, who might be interested in this info). I think the overhead of backend work and an added step in the release procedure for 3) are not worth it. So my preference is to keep this as is. Maybe we can do a light poll about this, since we have some differing opinions: **Please react to this comment with a 🎉 = option 1, 🚀 = option 2, 👀 = option 3**

File: docs/index.rst
Comment (Cadair) [reply docs/index.rst]: I am a little concerned that the contrast is very low on the readme in light mode. I don't think it would pass the accessibility contrast checker like it is at the moment. It certainly looks better in dark than light. (This is GH not docs fwiw)

**Comment by pllim** [comment]
I still think it is distracting and clutters the README but I will respect the majority vote.

**Comment by eerovaher** [comment]
Would the image be appropriate in our [contributing guidelines](CONTRIBUTING.md)?

**Comment by adrn** [comment]
@jeffjennings - it looks like GitHub markdown does support light/dark mode detection (e.g., [here](https://www.stefanjudis.com/notes/how-to-define-dark-light-mode-images-in-github-markdown/)), so could you use that feature here? Also, thinking about this again, I think the banner might make more sense down in the "Contributing" section instead of up at the top?

**Comment by jeffjennings** [comment]
48d6146 removes the stats image from the README; hopefully this resolves any remaining disagreement.

**Comment by jeffjennings** [comment]
> @jeffjennings - it looks like GitHub markdown does support light/dark mode detection (e.g., [here](https://www.stefanjudis.com/notes/how-to-define-dark-light-mode-images-in-github-markdown/)), so could you use that feature here? Also, thinking about this again, I think the banner might make more sense down in the "Contributing" section instead of up at the top?

I disagree that the image makes more sense in the Contributing section, but all of the above done in [8db4075](https://github.com/astropy/astropy/pull/16822/commits/8db40755f0fcf468a2d580aaa1865e3b06f1ed83)

File: docs/index.rst
Comment (jeffjennings) [reply docs/index.rst]: Thank you for noting this. [8db4075](https://github.com/astropy/astropy/pull/16822/commits/8db40755f0fcf468a2d580aaa1865e3b06f1ed83) resolves this by updating the image to match light or dark mode.

**Comment by pllim** [comment]
Does this also resolve #16560 ?

**Comment by adrn** [comment]
Given that the vote above was close, I asked @astropy/coordinators to advise on this PR. 

(FYI: The failing coverage check is an error and not relevant, I think)

**Comment by eteq** [comment]
FYI, @adrn @jeffjennings , the CoCo does have this on the agenda to advise as you asked, but just FYI it might be another ~week.

**Comment by pllim** [comment]
Thanks for your patience. Kelle took this CoCo agenda item "offline" and I think among all the CoCo, I might have the strongest opinion, so I will just make the executive decision now. Final tally is 10 🎉 (keep in readme) vs 5 🚀 (remove from readme) and none abstain. I want to note that all the 5 who voted to remove include both release manager, a website maintainer, and infrastructure support maintainers. That said, we are not doing weighted sum, so 10 still wins. I will squash merge now.

**Comment by jeffjennings** [comment]
Thank you for deciding and for letting me know. I hope it's a useful addition.

**Pull request merged**
